.buildkite/inference-benchmark-pipeline.sh

```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+set -euf
+
+# This script is executed by Buildkite to generate pipeline YAML string.
+# Document: https://buildkite.com/docs/pipelines/defining-steps#dynamic-pipelines
+#
+# Usage: CONVERT_RESULT_PATH=gs://path/to/result.tgz ./inference-benchmark-pipeline.sh
+
+# test cases to run inference
+# "<buildkite agent-type> <model filename>"
+TEST_CASES=(
+    "de10nano       lib/lib_arm.so"
+    "de10nano       lib/lib_fpga.so"
+
+    "jetson-nano    lib/lib_aarch64.so"
+    "jetson-nano    minimal_graph_with_shape.pb"
+
+    "jetson-tx2     lib/lib_aarch64.so"
+    "jetson-tx2     minimal_graph_with_shape.pb"
+
+    "jetson-xavier  lib/lib_aarch64.so"
+    "jetson-xavier  minimal_graph_with_shape.pb"
+
+    "raspberry-pi   lib/lib_aarch64.so"
+)
+
+# Generate setup step
+cat <<EOS
+steps:
+  - label: "inference: setup"
+    key: "setup"
+    timeout_in_minutes: "30"
+    agents:
+      - "env=benchmark"
+      - "agent-type=gcloudsdk"
+
+    artifact_paths:
+      - "convert-result.tgz"
+
+    command: |
+      gsutil -m cp "${CONVERT_RESULT_PATH}" ./
+EOS
+
+for TEST_CASE in "${TEST_CASES[@]}" ; do
+    IFS=' '
+    read AGENT MODEL <<< "${TEST_CASE}"
+
+    # Generate inference step
+    cat <<EOS
+
+
+  - label: "inference: ${AGENT} (${MODEL})"
+    depends_on: "setup"
+    timeout_in_minutes: "30"
+
+    agents:
+      - "env=benchmark"
+      - "agent-type=${AGENT}"
+
+    artifact_paths:
+      - "export/*/*/output/python/output/output.json"
+
+    command: |
+      buildkite-agent artifact download "convert-result.tgz" ./
+      tar xvf convert-result.tgz
+      cd export/*/*/output/python
+EOS
+
+    # Generate commands to run inference (with and without "sudo")
+    if [ "${AGENT}" = "de10nano" ] && [ "${MODEL}" = "lib/lib_fpga.so" ] ; then
```

Comment:
Why only fpga inference need sudo?
lib_arm.so dose need?

Reply:
I got an error message `permission denied, segmentation fault` without sudo.

This is because our binary for FPGA requires to access `/dev/mem`.
https://github.com/blue-oil/blueoil/blob/ce15ef3628af89a891dcf8a98f2af0174c7f0f18/dlk/python/dlk/templates/include/memdriver.h#L43-L54

And the code above is called only in the binary for FPGA.
https://github.com/blue-oil/blueoil/blob/ce15ef3628af89a891dcf8a98f2af0174c7f0f18/dlk/python/dlk/templates/src/network.tpl.cpp#L266-L267

Reply:
I see.